lua/telescope/builtin/files.lua

```diff
@@ -185,6 +189,82 @@ local function prepare_match(entry, kind)
   return entries
 end
 
+files.file_browser = function(opts)
+  opts = opts or {}
+
+  opts.cwd = opts.cwd and vim.fn.expand(opts.cwd) or vim.loop.cwd()
+
+  local gen_new_finder = function(path)
+    opts.cwd = path
+    local data = {}
+
+    scan.scan_dir(path, {
+      add_dirs = true,
+      depth = 1,
+      on_insert = function(entry, typ)
+        table.insert(data, typ == 'directory' and (entry .. os_sep) or entry)
+      end
+    })
+    table.insert(data, 1, '../')
+
+    return finders.new_table {
+      results = data,
+      entry_maker = (function()
+        local tele_path = require'telescope.path'
+        local gen = make_entry.gen_from_file(opts)
+        return function(entry)
+          local tmp = gen(entry)
+          tmp.ordinal = tele_path.make_relative(entry, opts.cwd)
+          return tmp
+        end
+      end)()
+    }
+  end
+
+  pickers.new(opts, {
+    prompt_title = 'Find Files',
+    finder = gen_new_finder(opts.cwd),
+    previewer = conf.file_previewer(opts),
+    sorter = conf.file_sorter(opts),
+    attach_mappings = function(prompt_bufnr, map)
+      actions._goto_file_selection:replace_if(function()
+        return actions.get_selected_entry().path:sub(-1) == os_sep
+      end, function()
+        local path = actions.get_selected_entry().path
+        actions.refresh(prompt_bufnr, gen_new_finder(vim.fn.expand(path:sub(1, -2))), { reset_prompt = true })
+      end)
+
+      local create_new_file = function()
+        local file = vim.fn.input('Create file or directory(add ' .. os_sep .. ' at the end of file). Enter name: ')
+        if file == "" then return end
+
+        local fpath = actions.get_selected_entry().path
+        if string.sub(fpath, -1) == os_sep then
+          fpath = vim.fn.fnamemodify(fpath, ':h:h')
+        else
+          fpath = vim.fn.fnamemodify(fpath, ':h')
+        end
+
+        fpath = fpath .. os_sep .. file
+
+        if string.sub(fpath, -1) ~= os_sep then
+          actions.close(prompt_bufnr)
+          Path:new(fpath):touch({ parents = true })
+          vim.cmd(string.format(':e %s', fpath))
+        else
+          Path:new(fpath:sub(1, -2)):mkdir({ parents = true })
+          actions.refresh(prompt_bufnr, gen_new_finder(vim.fn.expand(fpath)), { reset_prompt = true })
+        end
+      end
+
+      map('i', '<C-e>', create_new_file)
```

Comment:
Can I use this feature in not only `file_browser` but also `find_files` or `git_files`? I want to create a new file frequently in searching by that pickers.

And it is cool that the `Enter name:` prompt has the same words in Telescope prompt initially, I think.

Reply:
We can do this. Need to abstract some thinks to make this work. But i think the new `replace_mod` think can make this possible. Good point.

I was also thinking like doing somethink like this:
- Write some text in prompt buffer
- Now we have maybe entries maybe not
- So <C-e> doesn't open the selection but takes prompt text and creates new file.
- So basically getting rid of the vim.fn.input thingy

Reply:
@delphinus I am moving that sometimes else. It is now a really cool workflow but the fact that i am calling refresh makes it not usable more general. Maybe with enhanced. 

I need to think about this more but it needs to be merged soon because sunjon needs refresh and more api stuff that this PR provides.

Or you try it after its merged :)

Reply:
I think we could consider trying to do something like `actions.file_create` or something. I don't know how we  want to expose more of these actions to people so that they can be re-used easily.